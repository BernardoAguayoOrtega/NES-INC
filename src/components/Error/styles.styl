@import "../../stylus/variables.styl"

.container
  width: 100vw
  height: 100vh
  display: flex
  align-items: center
  justify-content: center
  flex-direction: column
  &__text
    font-family: Font1
    text-align: center
    color: darkGray
    font-weight: 700
    font-size: 8rem
  &__subtitle
    font-family: Font1
    text-align: center
    color: darkGray
    font-weight: 700
    font-size: 3rem


.container__animated
    animation-duration: 2.5s
    animation-fill-mode: both
    animation-iteration-count: infinite

@keyframes rubberBand
    0%
        transform: scale(1)
    30% 
        transform: scaleX(1.25) scaleY(0.75)
    40% 
        transform: scaleX(0.75) scaleY(1.25)
    60% 
        transform: scaleX(1.15) scaleY(0.85)
    100% 
        transform: scale(1)

.rubberBand
    animation-name: rubberBand