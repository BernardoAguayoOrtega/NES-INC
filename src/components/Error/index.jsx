import React from 'react';
import './styles.styl';

const Error = ({ message, subtitle }) => (
  <div className="container">
    <h1 className="container__text container__animated rubberBand">{message}</h1>
    <h2 className="container__subtitle">{subtitle}</h2>
  </div>
);

export default Error;
