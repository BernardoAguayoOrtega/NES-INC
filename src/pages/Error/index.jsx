/* This page is the one that is presented at the time of errors when finding a non-existent route or a service or server down */
// import react
import React from 'react';
// import styles
import './styles.styl';

// create and export error 404 page
const Error = ({ message, subtitle }) => (
  <div className="container">
    <h1 className="container__text container__animated rubberBand">{message}</h1>
    <h2 className="container__subtitle">{subtitle}</h2>
  </div>
);

export default Error
