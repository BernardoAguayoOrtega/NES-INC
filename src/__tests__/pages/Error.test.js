import React from 'react';
// import shallow
import { shallow } from 'enzyme';
// import app component
import Error from '../../pages/Error/index.jsx';

describe('<Error />', () => {
  const wrapper = shallow(<Error />);

  it('is render the Error?', () => {
    expect(wrapper.exists()).toBe(true);
  });

  it('is a container class?', () => {
    expect(wrapper.find('.container').exists()).toBe(true);
  });

  it('are there correct the title?', () => {
    expect(wrapper.find('h1').text()).toBe('');
  });

  it('are have two child?', () => {
    expect(wrapper.children().length).toBe(2);
  });
});
