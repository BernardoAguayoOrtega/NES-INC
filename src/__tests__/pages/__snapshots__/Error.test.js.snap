// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Error /> matches the snap shot 1`] = `
<div
  className="container"
>
  <h1
    className="container__text container__animated rubberBand"
  />
  <h2
    className="container__subtitle"
  />
</div>
`;
